Ryujinx/Ui/MainWindow.cs

```diff
@@ -0,0 +1,564 @@
+            FileChooserDialog fc = new FileChooserDialog("Choose the folder to open", this, FileChooserAction.SelectFolder, "Cancel", ResponseType.Cancel, "Open", ResponseType.Accept);
+
+            if (fc.Run() == (int)ResponseType.Accept)
+            {
+                LoadApplication(fc.Filename);
+            }
+
+            fc.Destroy();
+        }
+
+        private void Open_Ryu_Folder(object o, EventArgs args)
+        {
+            Process.Start(new ProcessStartInfo()
+            {
+                FileName = System.IO.Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "RyuFs"),
+                UseShellExecute = true,
+                Verb = "open"
+            });
+        }
+
+        private void Exit_Pressed(object o, EventArgs args) { End(); }
+
+        private void Window_Close(object o, DeleteEventArgs args) { End(); }
+
+        private void StopEmulation_Pressed(object o, EventArgs args)
+        {
+            // TODO: Write logic to kill running game
+        }
+
+        private void FullScreen_Toggled(object o, EventArgs args)
+        {
+            if (_fullScreen.Active == true) { Fullscreen();   }
+            else                            { Unfullscreen(); }
+        }
+
+        private void Settings_Pressed(object o, EventArgs args)
+        {
+            SwitchSettings SettingsWin = new SwitchSettings(_device);
+            _gtkapp.Register(GLib.Cancellable.Current);
+            _gtkapp.AddWindow(SettingsWin);
+            SettingsWin.Show();
+        }
+
+        private void Nfc_Pressed(object o, EventArgs args)
+        {
+            FileChooserDialog fc = new FileChooserDialog("Choose the file to open", this, FileChooserAction.Open, "Cancel", ResponseType.Cancel, "Open", ResponseType.Accept);
+            fc.Filter            = new FileFilter();
+            fc.Filter.AddPattern("*.bin");
+
+            if (fc.Run() == (int)ResponseType.Accept)
+            {
+                // TODO: Write logic to emulate reading an NFC tag
+            }
+            fc.Destroy();
+        }
+
+        private void Update_Pressed(object o, EventArgs args)
+        {
+            string ryuUpdater = System.IO.Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "RyuFS", "RyuUpdater.exe");
+            try { Process.Start(new ProcessStartInfo(ryuUpdater, "/U") { UseShellExecute = true }); }
+            catch(System.ComponentModel.Win32Exception) { CreateErrorDialog("Update canceled by user or updater was not found"); }
+        }
+
+        private void About_Pressed(object o, EventArgs args)
+        {
+            AboutWindow AboutWin = new AboutWindow();
+            _gtkapp.Register(GLib.Cancellable.Current);
+            _gtkapp.AddWindow(AboutWin);
+            AboutWin.Show();
+        }
+
+        private void Icon_Toggled(object o, EventArgs args)
+        {
+            if (_iconToggle.Active) SwitchSettings.SwitchConfig.GuiColumns[0] = true;
```

Comment:
```suggestion
            SwitchSettings.SwitchConfig.GuiColumns[0] = _iconToggle.Active;
```

And same for other `Toggled` method.